src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs

```diff
@@ -276,41 +275,43 @@ private static void PrepareCompilations(Solution solution, string projectPath, C
+                // Apply changes to all running projects, even if they do not have a static project dependency on any project that changed.
+                // The process may load any of the binaries using MEF or some other runtime dependency loader.
 
-            await ForEachProjectAsync(projectsToUpdate, async (runningProject, cancellationToken) =>
-            {
-                try
+                await ForEachProjectAsync(projectsToUpdate, async (runningProject, cancellationToken) =>
                 {
-                    using var processCommunicationCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(runningProject.ProcessExitedSource.Token, cancellationToken);
-                    var applySucceded = await runningProject.DeltaApplier.ApplyManagedCodeUpdates(updates.ProjectUpdates, processCommunicationCancellationSource.Token) != ApplyStatus.Failed;
-                    if (applySucceded)
+                    try
                     {
-                        runningProject.Reporter.Report(MessageDescriptor.HotReloadSucceeded);
-                        if (runningProject.BrowserRefreshServer is { } server)
+                        using var processCommunicationCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(runningProject.ProcessExitedSource.Token, cancellationToken);
+                        var applySucceded = await runningProject.DeltaApplier.ApplyManagedCodeUpdates(updates.ProjectUpdates, processCommunicationCancellationSource.Token) != ApplyStatus.Failed;
+                        if (applySucceded)
                         {
-                            runningProject.Reporter.Verbose("Refreshing browser.");
-                            await server.RefreshBrowserAsync(cancellationToken);
+                            runningProject.Reporter.Report(MessageDescriptor.HotReloadSucceeded);
+                            if (runningProject.BrowserRefreshServer is { } server)
+                            {
+                                runningProject.Reporter.Verbose("Refreshing browser.");
+                                await server.RefreshBrowserAsync(cancellationToken);
+                            }
                         }
                     }
-                }
-                catch (OperationCanceledException) when (runningProject.ProcessExitedSource.Token.IsCancellationRequested && !cancellationToken.IsCancellationRequested)
-                {
-                    runningProject.Reporter.Verbose("Hot reload canceled because the process exited.", emoji: "🔥");
-                }
-            }, cancellationToken);
-
+                    catch (OperationCanceledException) when (runningProject.ProcessExitedSource.Token.IsCancellationRequested && !cancellationToken.IsCancellationRequested)
+                    {
+                        runningProject.Reporter.Verbose("Hot reload canceled because the process exited.", emoji: "🔥");
```

Comment:
Is this a component where the images are constants?

Reply:
Not sure I understand. What images?